api/controllers/ClientController.js

```diff
@@ -206,14 +206,23 @@ module.exports = {
    *     description: Bad request. See response body for details.
    *   '403':
    *     description: Unauthorized. You are not an admin.
-   *#   '404':
-   *#     description: Requested client not found.
+   *   '404':
+   *     description: Requested client not found.
    */
   async destroy(request, reply) {
-    await Client.remove({ _id: request.params.id });
-    logger.info(
-      `[ClientController->destroy] Removed client ${request.params.id}`,
-    );
-    return reply.response().code(204);
+    const client = await Client.findOne({ _id: request.params.id });
+
+    // If a client was found, delete it and log the event.
+    if (client) {
+      await client.remove();
+
+      logger.info(
+        `[ClientController->destroy] Removed client ${request.params.id}`,
+      );
+      return reply.response().code(204);
+    }
+
+    // If no client was found, return 404.
+    return reply.response().code(404);
```

Comment:
Line 132: https://github.com/UN-OCHA/hid_api/blob/de716e4a04d819776110ba5b2102c144b68d0acf/api/controllers/ClientController.js#L132, `throw Boom.notFound()` is used to return a 404. Just wondering: why the difference?

Reply:
because I copied the line of code for the 204 and changed it 😁  I'll fix this up.